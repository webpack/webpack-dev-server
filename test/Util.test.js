'use strict';

const webpack = require('webpack');
const internalIp = require('internal-ip');
const Server = require('../lib/Server');
const createDomain = require('../lib/util/createDomain');
const config = require('./fixtures/simple-config/webpack.config');

describe('check utility funcitons', () => {
  let compiler;
  before(() => {
    compiler = webpack(config);
  });

  const tests = [{
    name: 'default',
    options: {
      host: 'localhost',
      port: 8080
    },
    expected: 'http://localhost:8080'
  }, {
    name: 'https',
    options: {
      host: 'localhost',
      port: 8080,
      https: true
    },
    expected: 'https://localhost:8080',
    timeout: 60000
  }, {
    name: 'override with public',
    options: {
      host: 'localhost',
      port: 8080,
      public: 'myhost.test'
    },
    expected: 'http://myhost.test'
  }, {
    name: 'override with public (port)',
    options: {
      host: 'localhost',
      port: 8080,
      public: 'myhost.test:9090'
    },
    expected: 'http://myhost.test:9090'
  }, {
    name: 'override with public (protocol)',
    options: {
      host: 'localhost',
      port: 8080,
      public: 'https://myhost.test'
    },
    expected: 'https://myhost.test'
  }, {
    name: 'override with public (protocol + port)',
    options: {
      host: 'localhost',
      port: 8080,
      public: 'https://myhost.test:9090'
    },
    expected: 'https://myhost.test:9090'
  }, {
    name: 'localIp',
    options: {
      useLocalIp: true,
      port: 8080
    },
    expected: `http://${internalIp.v4()}:8080`
  }];

  tests.forEach((t) => {
    const itInstance = it(`test createDomain '${t.name}'`, (done) => {
      const options = t.options;
      const server = new Server(compiler, options);
      const expected = t.expected;
      server.listen(options.port, options.host, (err) => {
        if (err) {
          done(err);
        }
        const generatedDomain = createDomain(options, server.listeningApp);
        if (generatedDomain !== expected) {
          done(`generated domain ${generatedDomain} doesn't match expected ${expected}`);
        } else {
          done();
        }
        server.close();
      });
    });
    if (t.timeout) {
      itInstance.timeout(t.timeout);
    }
  });
});
