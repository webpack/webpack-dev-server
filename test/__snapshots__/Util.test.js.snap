// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`findPort cli utility function should throw the error when the port isn't found 1`] = `"No open ports found in between 8080 and 8085"`;
